Locale bug?! FAIL: TestGetFirstEntry: rsl_test.go:984: unable to read reference 'refs/gittuf/reference-state-log': exit status 128 ... révision inconnue ou chemin inexistant.

### What happened?

Hi!  I'm maintaining this package in Debian.  Debian attempts to build packages in "strange" environment to trigger reproducability bug.  This triggers a self-test failure in gittuf.  Is it trying to parse `git` output perhaps?  Then it should invoke it like `env LC_ALL=C git ...` for predictable outputs.  Maybe the problem is something else though.

https://salsa.debian.org/jas/gittuf/-/jobs/7260564

```
    --- PASS: TestParseRSLEntryText/propagation_entry,_non-zero_commit (0.00s)
    --- PASS: TestParseRSLEntryText/entry,_non-zero_commit (0.00s)
=== CONT  TestGetFirstEntry
    rsl_test.go:984: unable to read reference 'refs/gittuf/reference-state-log': exit status 128 when executing `git --git-dir /tmp/TestGetFirstEntry523787633/001/.git rev-parse refs/gittuf/reference-state-log`: fatal : argument 'refs/gittuf/reference-state-log' ambigu : révision inconnue ou chemin inexistant.
        Utilisez '--' pour séparer les chemins des révisions, comme ceci :
        'git <commande> [<révision>...] -- [<chemin>...]'
        
--- FAIL: TestGetFirstEntry (0.03s)
FAIL
FAIL	github.com/gittuf/gittuf/internal/rsl	1.015s
?   	github.com/gittuf/gittuf/internal/signerverifier/common	[no test files]
=== RUN   TestCreateEnvelope
--- PASS: TestCreateEnvelope (0.00s)
=== RUN   TestSignEnvelope
--- PASS: TestSignEnvelope (0.03s)
```

Thanks,
Simon

### Relevant log output

```Shell

```

### Code of Conduct

- [x] I agree to follow this project's Code of Conduct

--- a/.github/workflows/ci-locale.yml
+++ b/.github/workflows/ci-locale.yml
@@ -1,0 +1,36 @@
+name: tests-locale
+on:
+  push:
+    branches: ['main']
+    paths-ignore:
+      - "docs/*"
+      - "**.md"
+  pull_request:
+    paths-ignore:
+      - "docs/*"
+      - "**.md"
+permissions: read-all
+jobs:
+  test:
+    strategy:
+      matrix:
+        go-version: ['1.23', '1.24']
+    runs-on: ubuntu-latest
+    steps:
+    - name: Set locale
+      run: |
+        sudo apt-get update && sudo apt-get install tzdata locales -y
+        sudo sed -i '/fr_FR.UTF-8/s/^# //g' /etc/locale.gen && sudo locale-gen
+    - name: Checkout code
+      uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683
+    - name: Install Go
+      uses: actions/setup-go@f111f3307d8850f501ac008e886eec1fd1932a34
+      with:
+        go-version: ${{ matrix.go-version }}
+        cache: true
+    - name: Test
+      run: go test -timeout 20m ./...
+      env:
+        LANG: "fr_FR.UTF-8"
+        LANGUAGE: "fr_FR:fr"
+        LC_ALL: "fr_FR.UTF-8"

--- a/internal/gitinterface/repository.go
+++ b/internal/gitinterface/repository.go
@@ -139,6 +139,7 @@ func (e *executor) execute() (io.Reader, io.Reader, error) {
 	}
 	cmd := exec.Command(binary, e.args...) //nolint:gosec
 	cmd.Env = e.env
+	cmd.Env = append(cmd.Env, "LC_ALL=C") // force git to the C (and thus english) locale
 
 	var (
 		stdOut bytes.Buffer

